src/build/app/mod.rs

```diff
@@ -1461,6 +1461,36 @@ impl<'b> App<'b> {
         debug_assert!(self._app_debug_asserts());
         self.args._build();
         self.settings.set(AppSettings::Built);
+        Self::_panic_on_missing_help(&self, self.g_settings.is_set(AppSettings::HelpRequired));
+    }
+
+    #[cfg(not(debug_assertions))]
+    fn _panic_on_missing_help(app: &App, help_required_globally: bool) {}
+
+    #[cfg(debug_assertions)]
+    fn _panic_on_missing_help(app: &App, help_required_globally: bool) {
+        for sub_app in &app.subcommands {
+            Self::_panic_on_missing_help(&sub_app, help_required_globally);
+        }
+        Self::_panic_on_missing_help_driver(&app, help_required_globally);
+    }
+
+    #[cfg(debug_assertions)]
+    fn _panic_on_missing_help_driver(app: &App, help_required_globally: bool) {
```

Comment:
You don't need this function, just merge it into the one above.

Reply:
Agreed.